src/main/java/uk/gov/pay/api/app/PublicApi.java

```diff
@@ -41,13 +46,24 @@ public void initialize(Bootstrap<PublicApiConfig> bootstrap) {
     public void run(PublicApiConfig config, Environment environment) throws Exception {
         final Client client = RestClientFactory.buildClient(config.getRestClientConfig());
 
-        configureObjectMapper(config, environment.getObjectMapper());
+        ObjectMapper objectMapper = environment.getObjectMapper();
+        configureObjectMapper(config, objectMapper);
 
         environment.healthChecks().register("ping", new Ping());
+        environment.jersey().register(new HealthCheckResource(environment));
 
         environment.jersey().register(new PaymentsResource(client, config.getConnectorUrl()));
-        environment.jersey().register(new HealthCheckResource(environment));
+
+        RateLimiter rateLimiter = new RateLimiter(config.getRateLimiterConfig().getRate(), config.getRateLimiterConfig().getPerMillis());
+
+        environment.servlets().addFilter("AuthorizationValidationFilter", new AuthorizationValidationFilter(config.getApiKeyHmacSecret()))
+                .addMappingForUrlPatterns(of(REQUEST), true, "/v1/*");
+
+        environment.servlets().addFilter("RateLimiterFilter", new RateLimiterFilter(rateLimiter, objectMapper))
+                .addMappingForUrlPatterns(of(REQUEST), true, "/v1/*");
```

Comment:
[PP-693](https://payments-platform.atlassian.net/browse/PP-693)
Have a look to check if the explanation is clear enough.